runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -262,10 +263,19 @@ class ExecutionContext final {
     std::unique_ptr<MemoryManager> memory_man_ptr{nullptr};
     std::unique_ptr<PythonInterpreterGuard> py_guard{nullptr};
 
+    // PRNG
+    uint32_t *seed = nullptr;
```

Comment:
```suggestion
    uint32_t *seed;
```
Not a big deal but this provides a default in two places (here and in the constructor).

Reply:
let's keep this one and eliminate the one in the constructor, since the constructor always receives a seed from 
```
void __catalyst__rt__initialize(uint32_t *seed)
{
    CTX = std::make_unique<ExecutionContext>(seed);
}
```
and thus doesn't need a default. 

This default is here because otherwise wheels build will fail to address sanitizer complaining about uninitialized pointer attributes in a class

Reply:
I think the constructor default does have the benefit of requiring to provide a seed value though, which is much more convenient. Have you tried using the initializer list like for the other parameter? Maybe that would silence the sanitizer since it's guaranteed to happen.

Reply:
yes, I noticed not having the constructor default means modifying more tests where execution context is created

I haven't tried initilizer list (also doesn't the initializer list also provide the default too?); I'll leave it as it if we're fine with it. The only "downside" is one assignment of nullptr to something that's nullptr already

Reply:
Yup, more of a "single source of truth" issue (changing the default requires changing two places.